pkg/apis/apps.kubermatic/v1/application_definition.go

```diff
@@ -57,25 +64,50 @@ type GitCredentials struct {
 	// Authentication method
 	Method GitAuthMethod `json:"method"`
 
-	// Username holds the ref and key in the secret for the username credential
-	Username *GlobalSecretKeySelector `json:"username,omitempty"`
+	// Username holds the ref and key in the secret for the username credential. Secret must exist in the namespace where
+	// KKP is installed.
+	Username *corev1.SecretKeySelector `json:"username,omitempty"`
 
-	// Password holds the ref and key in the secret for the Password credential
-	Password *GlobalSecretKeySelector `json:"password,omitempty"`
+	// Password holds the ref and key in the secret for the Password credential. Secret must exist in the namespace where
+	// KKP is installed.
+	Password *corev1.SecretKeySelector `json:"password,omitempty"`
```

Comment:
Can we reuse `HelmCredentials` here and move Token, SSHKey to that struct as well?

Reply:
as discussed, the requirements are different. For  Helm, credentials are required, for git they are optional.